ddspipe_core/include/ddspipe_core/core/DdsPipe.hpp

```diff
@@ -308,6 +347,13 @@ class DdsPipe
      */
     std::map<utils::Heritable<types::DistributedTopic>, bool> current_topics_;
 
+    /**
+     * @brief List of the ids of the participants who discovered each topic.
+     *
+     * Every topic discovered is added to the map.
+     */
+    std::map<utils::Heritable<types::DistributedTopic>, types::ParticipantId> current_topics_discoverers_;
```

Comment:
I like the idea of adding the discoverer_participant_id to the Topic objects. We don't have to worry about the Topic objects being the keys to maps, because the Topic objects override the == operator.

Reply:
Cool, I will not resolve this yet as I need to double check the key thing is not an issue.